wireapp/wire-ios-mono
# fix: remove filtering events - WPB-9221

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-9221" title="WPB-9221" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-9221</a>  Missing messages during sync
  </td></table>
  <br />
 









### Issue

Here's a fix for a potential scenario for missing messages:
1. The app fetches 100 events 
2. It clears the already fetch event ids before we actually processEvents from ZMMissingUpdateEventsTranscoder 
3. it stores 100 event ids and then decrypt and store up to 50 events (storing the 50th event as lastUpdateEventId)
4. When opening the app, it would start to fetch a new batch of events from last update event (51...), processing the filtered events(51th->100th events) previously fetched


##### Solution:

For now it's easier to remove the filtering of already fetched events, which would resolve the current issue. It will produce decryption errors for duplicate messages but these errors won't make it to the user. 

Note: it would be easier to reimplement a filtering solution if needed after we refactored the quick sync.


### Testing

1. add delay to decryptAndStoreEvents
2. spam 100 events via background fetch or via notification extension
3. let background fetch happen 
4. check we don't have missing events
---

### Checklist
